sound/soc/sof/intel/hda-dsp.c

```diff
@@ -239,3 +239,53 @@ int hda_dsp_core_reset_power_down(struct snd_sof_dev *sdev,
 	return ret;
 }
 
+int hda_dsp_suspend(struct snd_sof_dev *sdev, int state)
+{
```

Comment:
I'd expect most of this function to be in pm.c as generic code. The HW specific part hda_dsp_core_reset_power_down() can be an HW abstracted operation (add ops to the operations callback structure). e.g. we could have snd_sof_dsp_suspend(sdev), snd_sof_dsp_resume(sdev)

Reply:
@plbossart @lgirdwood I have updated this to use the chip cores_mask to make it generic.